frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedRows.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { t } from "ttag";
+
+import { ChartSettingOrderedItems } from "./ChartSettingOrderedItems";
+
+import {
+  ChartSettingMessage,
+  ChartSettingOrderedRowsRoot,
+} from "./ChartSettingOrderedRows.styled";
+
+interface Row {
+  rowIndex: number;
+  display_name: string;
+  enabled: boolean;
+}
+
+interface ChartSettingOrderedRowsProps {
+  onChange: (rows: Row[]) => void;
+  rows: Row[];
+  value: Row[];
+}
+
+export const ChartSettingOrderedRows = ({
+  onChange,
+  rows,
+  value,
+}: ChartSettingOrderedRowsProps) => {
+  const handleDisable = (row: any) => {
+    const rowsCopy = [...value];
+    const index = rowsCopy.findIndex((r: Row) => r.rowIndex === row.rowIndex);
+    rowsCopy[index] = { ...rowsCopy[index], enabled: !row.enabled };
+    onChange(rowsCopy);
```

Comment:
What do you think about using [icepick's `updateIn`](https://github.com/aearly/icepick#updateincollection-path-callback), so you don't have to create a new array copy and mutate it locally. And possibly make this a bit more readable (could be subjective, but I do find this looks more declarative)?

```suggestion
    const index = value.findIndex((r: Row) => r.rowIndex === row.rowIndex);
    onChange(
      updateIn(value, [index], row => ({ ...row, enabled: !row.enabled })),
    );
```

Reply:
I could give it a try! In this case, I was just following the pattern that was already in `<ChartSettingOrderedColumns />`, but I'll see if there are any weird side effects